subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -709,6 +710,30 @@ public void applyTo(String value, StartParameterInternal settings, Origin origin
             settings.setConfigurationCacheHeapDumpDir(value);
         }
     }
+
+    /**
+     * Whether [project property accesses][org.gradle.api.internal.properties.GradleProperties] are tracked individually
+     * to increase cache hit rates.
+     *
+     * Increases memory usage proportionally to the number of projects and property accesses.
+     *
+     * It can be disabled to save on memory.
+     *
+     * The default is `true`.
+     */
+    public static class ConfigurationCacheFineGrainedPropertyTracking extends BooleanBuildOption<StartParameterInternal> {
+        public static final String PROPERTY_NAME = "org.gradle.configuration-cache.fine-grained-property-tracking";
```

Comment:
❓  Does this have to be another public property on the start parameter?

Reply:
In order to allow init script based configuration? I guess so 🤔

Reply:
> Why do we go with a stable public property instead of an internal flag?

This is intended as supported behavior. We recognize there's overhead (however small we can make it, it would still be proportional to the number of property accesses times the number of projects), and we want to allow users to opt-out when it makes sense. For example, when the overhead becomes prohibitive or simply provides no benefits.